src/pipelines/rust_app.rs

```diff
@@ -291,26 +292,59 @@ pub struct RustAppOutput {
     pub wasm_output: String,
 }
 
+pub fn pattern_evaluate(pattern: String, params: &HashMap<String, String>) -> String {
+    let mut result = pattern.clone();
+    for (k, v) in params.iter() {
+        let needle = "{".to_owned() + k.as_str() + "}";
```

Comment:
```suggestion
        let pattern = format!("{{{}}}", k.as_str());
```
- Playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e1a70746c34948f89d6573be833ef61a
- Docs: https://doc.rust-lang.org/std/fmt/#escaping